Neos.Flow/Classes/ObjectManagement/Configuration/ConfigurationBuilder.php

```diff
@@ -353,6 +373,51 @@ protected function parseArgumentOfTypeObject($argumentName, $objectNameOrConfigu
         return $argument;
     }
 
+    /**
+     * Creates a "virtual object configuration" for object arguments, turning:
+     *
+     * 'Some\Class\Name':
+     *   arguments:
+     *     1:
+     *       object:
+     *         factoryObjectName: 'Some\Factory\Class'
+     *
+     * into:
+     *
+     * 'Some\Class\Name':
+     *   arguments:
+     *     1:
+     *       object: 'Some\Class\Name:argument:1'
+     *
+     * 'Some\Class\Name:argument:1':
+     *   factoryObjectName: 'Some\Factory\Class'
+     *
+     *
+     * @param array &$objectConfigurations
+     * @return void
+     * @throws InvalidObjectConfigurationException
+     */
+    protected function wireObjectArguments(array &$objectConfigurations)
+    {
+        /** @var Configuration $objectConfiguration */
+        foreach ($objectConfigurations as $objectConfiguration) {
+            /** @var ConfigurationArgument $argument */
+            foreach ($objectConfiguration->getArguments() as $index => &$argument) {
+                if ($argument === null || $argument->getType() !== ConfigurationArgument::ARGUMENT_TYPES_OBJECT) {
+                    continue;
+                }
+                $argumentValue = $argument->getValue();
+                if (!$argumentValue instanceof Configuration) {
+                    continue;
+                }
+                $argumentObjectName = $objectConfiguration->getObjectName() . ':argument:' . $index;
+                $argumentValue->setObjectName($argumentObjectName);
+                $objectConfigurations[$argumentObjectName] = $argument->getValue();
+                $argument->set($argument->getIndex(), $argumentObjectName, $argument->getType());
```

Comment:
Ah, psalm now doing the good job again:
```
Argument 1 of Neos\Flow\ObjectManagement\Configuration\ConfigurationArgument::set expects int, string provided
$argument->set($argument->getIndex(), $argumentObjectName, $argument->getType());
```
https://travis-ci.org/github/neos/flow-development-collection/jobs/670022547#L796
:)

Reply:
Ah right, I forgot about that one.
It's great, that `getIndex()` returns a string and `set` expects an int.. but that's not the only weird part about that ancient code :)
fixed